src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
+    public MyLinkedList() {
+        tail = null;
+        head = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        if (tail == null) {
+            tail = head = new Node<>(null, value, null);
+        } else if (index == 0) {
+            tail.prev = new Node<>(null, value, tail);
+            tail = tail.prev;
+        } else if (index == size) {
+            head.next = new Node<>(head, value, null);
+            head = head.next;
+        } else {
+            Node<T> current = tail;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+            current.prev.next = new Node<>(current.prev, value, current);
+            current.prev = current.prev.next;
+        }
+        size++;
```

Comment:
you need to divide logic from here into different methods and optimize some code. Small hint: if `index == size`, just call method add(T value).